packages/core-manager/src/workers/actions/streams/iterator-to-stream.ts

```diff
@@ -0,0 +1,12 @@
+import { Readable } from "stream";
+
+export class IteratorToStream extends Readable {
+    public constructor(private readonly iterator: IterableIterator<any>) {
+        super({ objectMode: true });
+    }
+
+    public _read(size: number) {
+        const item = this.iterator.next();
+        this.push(item.done ? null : item.value);
+    }
+}
```

Comment:
There is standard [`Readable.from`] method that creates readable stream from iterator.

[`Readable.from`]: https://nodejs.org/api/stream.html#stream_stream_readable_from_iterable_options